src/Time/Actions/ChangeStepSize.hpp

```diff
@@ -105,15 +129,34 @@ struct ChangeStepSize {
   template <typename DbTags, typename... InboxTags, typename Metavariables,
             typename ArrayIndex, typename ActionList,
             typename ParallelComponent>
-  static std::tuple<db::DataBox<DbTags>&&> apply(
+  static std::tuple<db::DataBox<DbTags>&&, bool, size_t> apply(
       db::DataBox<DbTags>& box, tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
       const Parallel::GlobalCache<Metavariables>& cache,
       const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
       const ParallelComponent* const /*meta*/) noexcept {
     static_assert(Metavariables::local_time_stepping,
                   "ChangeStepSize can only be used with local time-stepping.");
-    change_step_size<StepChooserRegistrars>(make_not_null(&box), cache);
-    return std::forward_as_tuple(std::move(box));
+    const bool step_successful =
+        change_step_size<StepChooserRegistrars>(make_not_null(&box), cache);
+    if (step_successful) {
+      return {std::move(box), false,
+              tmpl::index_of<ActionList, ChangeStepSize>::value};
+    } else {
+      if constexpr (std::is_same_v<
+                        tmpl::index_if<
+                            ActionList,
+                            tt::is_a_lambda<Actions::UpdateU, tmpl::_1>>,
+                        tmpl::no_such_type_>) {
```

Comment:
`tmpl::none<ActionList, tt::is_a_lambda<Actions::UpdateU, tmpl::_1>>` or store the result of the `index_if` and use it again below.